addons/pos_self_order/static/src/common/hooks/use_detection.js

```diff
@@ -31,11 +31,7 @@ export function useDetection(root, targets, getDeps) {
                         delete allVisible[name];
                     }
 
-                    if (Object.keys(allVisible).length > 0) {
-                        selectedName = Object.entries(allVisible).sort((a, b) => {
-                            return b[1] - a[1];
-                        })[0][0];
-                    }
+                    selectedName = Object.keys(allVisible).reverse()[0];
```

Comment:
Maybe instead of using the intersection ratio, we select the first entry appearing ? This is just a recommendation that we can discuss about but for me it makes more sense.